src/lib/nav/NavigationRailItem.svelte

```diff
@@ -0,0 +1,175 @@
+<script lang="ts">
+  import type { IconifyIcon } from "@iconify/types";
+
+  import Layer from "$lib/misc/Layer.svelte";
+  import Icon from "$lib/misc/_icon.svelte";
+
+  let {
+    label,
+    icon,
+    active = false,
+    badge,
+    onclick,
+  }: {
+    label: string;
+    icon: IconifyIcon;
+    active?: boolean;
+    badge?:
+      | {
+          label?: string;
+          small?: boolean;
+        }
+      | true;
+    onclick?: (e: MouseEvent) => void;
+  } = $props();
+</script>
+
+<button class="m3-container" class:active {onclick}>
+  <Layer />
+
+  <div class="icon">
+    <Icon {icon} />
+
+    {#if badge}
+      <span class="badge" class:small={badge && typeof badge === "object" ? badge.small : false}>
+        {typeof badge === "object" ? badge.label : ""}
+      </span>
+    {/if}
```

Comment:
i don't like only having badges on one component, they should work for any element with Icons (you could probably write a function to add badges that takes in an IconifyIcon and returns an IconifyIcon with badge code appended to the end of the `body`) or not be here